### PR TITLE
Copy Prosal Implemented

### PR DESCRIPTION
**Pull request by OscarMarzo**
Pull request Copy Proposal

### CODE REVIEW

File: client/src/components/CollapsibleTable.jsx [hunk 1]
```diff
@@ -109,6 +110,13 @@ function Row(props) {
             <DeleteIcon />
           </IconButton>
         </TableCell>
+        <TableCell style={{ width: '3%' }}> 
+        <Link to={`/teacher/copyProposal/${row.p.id}`}>
+          <IconButton aria-label="copy" onClick={() => { }} disabled={proposalAccepted  || row.p.status === 'assigned'}>
```
Comment (filippoCaste) [inline client/src/components/CollapsibleTable.jsx:115]: What does this onClick function do?

Comment (OscarMarzo) [reply client/src/components/CollapsibleTable.jsx:115]: When the teacher click on IconButton he will be send to the route /teacher/copyProposal/:proposalId where there is the form of the proposal and the can add this

Comment (filippoCaste) [reply client/src/components/CollapsibleTable.jsx:115]: For compatibility and coherence reasons, don't use Link to, and use Navigate.

You need to import it and use as follows:
````js
const navigate = useNavigate();
<IconButton ... onClick={ () => navigate('...');} ...>
```

File: client/src/components/ProposalTeacher.jsx [hunk 2]
```diff
@@ -156,138 +132,259 @@ function AddProposalTeacher(props)
...
-       if(corretto == false) 
-       {
-           handleMessage("ATTENTION: "+campi_vuoti+" EMPTY ", "warning");
-       }
+    //ADD PROPOSAL  
+    if(typeOperation=="add")
+    {
+      proposalAPI.postProposal(title,type,description,level, formatted_expiration,notes,
+          required_knowledge, array_only_cod_degree,cod_group,supervisors_obj,selectedKeywordList)
+          .then(navigate("/teacher"))
```
Comment (filippoCaste) [inline client/src/components/ProposalTeacher.jsx:336]: Missing `() => navigate('/teacher')`

**Review by filippoCaste — APPROVED** [review]
done, regexp can create issues

**Pull request merged**